### PR TITLE
fix(smart-field): log error on missing parameter dependencies and avoid crash

### PR DESCRIPTION
**Pull request by SkyHuss**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by SkyHuss** [comment]
Task linked: [CU-860rzeuat 🔥🔥 P2 - Collection customization - It crashes if dependencies of a smart field is missing](https://app.clickup.com/t/860rzeuat)

File: packages/datasource-customizer/src/collection-customizer.ts [hunk 1]
```diff
@@ -183,22 +183,32 @@ export default class CollectionCustomizer<
...
-          return false;
-        }
-      });
 
-      const collection = canBeComputedBeforeRelations
-        ? collectionBeforeRelations
-        : collectionAfterRelations;
-
-      collection.registerComputed(name, mapDeprecated<S, N>(definition));
+      if (definition.dependencies) {
```
Comment (ghusse) [inline packages/datasource-customizer/src/collection-customizer.ts:190]: Can you apply the "return early" pattern? Meaning that it's preferable to test for the absence of `dependencies`, log something and then return.

It's easier to read to not have a lot of code inside a `if/else` section.

File: packages/datasource-customizer/src/collection-customizer.ts [hunk 2]
```diff
@@ -183,22 +183,32 @@ export default class CollectionCustomizer<
...
+        });
+
+        const collection = canBeComputedBeforeRelations
+          ? collectionBeforeRelations
+          : collectionAfterRelations;
+
+        collection.registerComputed(name, mapDeprecated<S, N>(definition));
+      } else {
+        logger(
+          'Error',
```
Comment (ghusse) [inline packages/datasource-customizer/src/collection-customizer.ts:206]: Is it an error or a warning in this case?

File: packages/datasource-customizer/test/collection-customizer.test.ts [hunk 3]
```diff
@@ -347,6 +347,26 @@ describe('Builder > Collection', () => {
...
+      const loggerMock = jest.fn();
+
+      const fieldDefinition: ComputedDefinition = {
+        columnType: 'String',
+        getValues: records => records.map(() => 'aaa'),
+      };
+
+      customizer.addField('no_dependency_field', fieldDefinition);
+
+      await dsc.getDataSource(loggerMock);
```
Comment (ghusse) [inline packages/datasource-customizer/test/collection-customizer.test.ts:362]:
```suggestion
      await expect(dsc.getDataSource(loggerMock)).toResolve();
```

File: packages/datasource-customizer/src/collection-customizer.ts (hunk 2 above)
Comment (SkyHuss) [reply packages/datasource-customizer/src/collection-customizer.ts:206]: i think it's an error because it's not suppose to append ?

Comment (ghusse) [reply packages/datasource-customizer/src/collection-customizer.ts:192]: Yes, but the agent does not crash. So it seems to work properly even with this "error". Do we have other error logs that are not stopping the agent?

**Review by ghusse — APPROVED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✖️&nbsp; Integration tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;Destination branch
- ✅&nbsp;PR linked to the clickup task

**Pull request merged**